pkg/collectors/destinationrules.go

```diff
@@ -0,0 +1,265 @@
+package collectors
+
+import (
+	"context"
+	"fmt"
+	"github.com/douglas-reid/istio-state-metrics/pkg/apis/networking/v1alpha3"
+	"github.com/douglas-reid/istio-state-metrics/pkg/client/clientset/versioned"
+	informers "github.com/douglas-reid/istio-state-metrics/pkg/client/informers/externalversions/networking/v1alpha3"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/tools/cache"
+	"strings"
+)
+
+var (
+	descDestinationRulesInfo = prometheus.NewDesc(
+		"istio_pilot_destination_rule_info",
+		"Information about Pilot DestinationRules",
+		[]string{"DestinationRule", "namespace"},
+		nil,
+	)
+
+	descDestinationRulesHost = prometheus.NewDesc(
+		"istio_pilot_destination_rule_host",
+		"Information about Host in Pilot DestinationRules",
+		[]string{"DestinationRule", "host"},
+		nil,
+	)
+
+	descDestinationRulesTrafficPolicyLoadBalancer = prometheus.NewDesc(
+		"istio_pilot_destination_rule_traffic_policy_loadbalancer",
+		"Information about LoadBalancer in Pilot DestinationRules",
+		[]string{"DestinationRule", "lb_type", "lb_identifier", "consistenthash_minimumringsize"},
+		nil,
+	)
+
+	descDestinationRulesTrafficPolicyConnectionPoolSettings = prometheus.NewDesc(
+		"istio_pilot_destination_rule_traffic_policy_connection_pool_settings",
+		"Information about ConnectionPoolSettings in Pilot DestinationRules",
+		[]string{"DestinationRule", "tcp_maxconnections", "tcp_connecttimeout", "http_http1MaxPendingRequests",
```

Comment:
curious: does it make sense to split these out at all into `connectionpool_tcpsettings` and `connectionpool_httpsettings` ?

Reply:
Done :)